src/main/java/hudson/plugins/warnings/WarningsDescriptor.java

```diff
@@ -43,7 +42,23 @@
     /** Icon to use for the result summary. */
     public static final String LARGE_ICON_URL = IMAGE_PREFIX + "warnings-48x48.png";
 
-    private final CopyOnWriteList<GroovyParser> groovyParsers = new CopyOnWriteList<GroovyParser>();
+    /** Backwards compatibility */
+    private transient CopyOnWriteList<GroovyParser> groovyParsers;
```

Comment:
CopyOnWriteList has been created for a reason - to prevent issues on concurrent access. This change would nuke the plugin

Reply:
Thank @oleg-nenashev I'll see if i can figure out how to use the same data type.

Reply:
So if i understand the issue correctly the concurrency issue arises when more than 1 user configures it at the same time? Is there some historical reason as to why this is handled differently in the warnings plugin than in any other plugin? AFAIK most other plugin's just do a full replace on each save, so the guy to save last always wins in the case they started editing at the same time.

Reply:
No, write operation may collide with Read operations. So any reconfiguration may break plugin usages in builds happening in parallel with it

Reply:
@oleg-nenashev Ah okay that explains it. I'll see if i can figure out how to solve this with the new configure method and the setter.